### ISSUE TITLE
Cache-Control

### ISSUE THREAD

**Issue by tarekziade** [Kinto/kinto-changes#13]
It looks like Cache-Control is set to no-cache - where we need to cache this collection

I propose to set the cache_expires option for this collection to 1 hour by default, and expose it as an option of the plugin.

**Comment by leplatrem** [comment]
This is one way of solving one part of [our CDN caching problem](https://bugzilla.mozilla.org/show_bug.cgi?id=1273896), but in my opinion this is unrelated to this plugin.

* We'll have the same necessity to add cache control on other collections of records
* The cache setting has no reason to be set if the instance is not behing a CDN

Instead, I would add the following setting in the readonly instance that's behind the CDN:

```
kinto.record_cache_expires_seconds = 3600
```

This way, every record resource is cached properly, whether it's the /changes endpoint or not

**Comment by tarekziade** [comment]
I think however that this collection is very specific. It's called all the time by firefox prior to calling any other collections. We probably don't want to cache it like others.

+1 for setting a cache for all collections, but also so +1 for a specific option here.

**Comment by leplatrem** [comment]
> We probably don't want to cache it like others.

Good news, [according to the docs](http://kinto.readthedocs.io/en/stable/configuration/settings.html#client-caching) we can control cache by collection from settings:

```ini
kinto.record_cache_expires_seconds = 60
kinto.monitor_changes_record_cache_expires_seconds = 3600
```

![](http://i.giphy.com/yF0D8vvREn4TS.gif)

> +1 for a specific option here
This is trivial to implement, but if existing settings can do the job, we may want to avoid it no?

**Comment by tarekziade** [comment]
> if existing settings can do the job, we may want to avoid it no?

I did not know we could do this, so this is great news !

That said, the goal of the "kinto-changes" project is not only to feed automatically a collection when things changes - and let the admin set several options. The goal is to provide a "changes" API that works out of the box.

So when I deploy that plugin, I expect it to have decent default values for everything required to have a "changes API" properly configured, and that includes the cache settings imho.

In other words, it's great to be able to tweak everything like you showed, but it's also cool if we can just have it working out of the box, and have the simplest configuration possible when we want to change this behavior in Kinto: "Let's cache in Kinto for 2 hours what triggers an update of *all* the data in Firefox"

So maybe a first step would be to make sure we have decent default values for that collection cache.

**Comment by tarekziade** [comment]
btw just a nit

```ini
kinto.monitor_changes_record_cache_expires_seconds
```
I find this settings complex to read, it's not obvious where are the bucket and collection names.
(Good luck if the collection is named cache ;))

Since the caching settings are one logical unit, what about using namespaces to make things clearer ?

```ini
kinto.cache.expires_seconds:default = 3600
kinto.cache.expires_seconds:monitor = 360
kinto.cache.expires_seconds:monitor.changes.record = 60
```

**Comment by tarekziade** [comment]
I hate ini files restrictions..... I whish we'd use json or yaml  :)

**Comment by leplatrem** [comment]
> The goal is to provide a "changes" API that works out of the box

I know what you mean, and we could easily set a default value for this setting when the plugin is enabled. One-liner.

I don't want to block the initiative if everybody agrees! Just trying to explain my point of view :)

But I think that the changes collection does not have to be cached. It's more related to our setup (readonly, CDN etc.).

Imagine (*I know, we should focus on concrete use-cases*) that this changes endpoint is used the kinto-admin to reflect the last update of the different collections. Wouldn't we want to refresh it immediately when something is saved?

> I find this settings complex to read,

+1000

**Comment by leplatrem** [comment]
#stalled
